x-pack/plugins/security_solution/public/common/components/exceptions/viewer/index.tsx

```diff
@@ -4,96 +4,411 @@
+        />
+      );
+    } else if (filterOptions.showDetectionsList) {
+      return (
+        <FormattedMessage
+          id="xpack.siem.exceptions.viewer.exceptionDetectionDetailsDescription"
+          defaultMessage="All exceptions to this rule are applied to the detection rule, not the endpoint. View {ruleSettings} for more details."
+          values={{
+            ruleSettings: (
+              <EuiLink href={ruleSettingsUrl} target="_blank">
+                <FormattedMessage
+                  id="xpack.siem.exceptions.viewer.exceptionDetectionDetailsDescription.ruleSettingsLink"
+                  defaultMessage="rule settings"
+                />
+              </EuiLink>
+            ),
+          }}
+        />
+      );
+    } else {
+      return <></>;
+    }
+  }, [filterOptions.showEndpointList, filterOptions.showDetectionsList, ruleSettingsUrl]);
+
+  const showEmpty = useMemo((): boolean => {
+    return !initLoading && !loadingList && exceptions.length === 0;
+  }, [initLoading, exceptions.length, loadingList]);
 
   return (
-    <EuiPanel paddingSize="none">
-      <EuiFlexGroup direction="column" gutterSize="none">
-        <EuiFlexItem>
-          <EuiFlexGroup direction="row">
-            <ExceptionDetails
-              showComments={showComments}
-              exceptionItem={exceptionItem}
-              onCommentsClick={onCommentsClick}
+    <>
+      {isModalOpen && (
+        <EuiOverlayMask>
+          <EuiModal onClose={onCloseExceptionModal}>
+            <EuiModalBody>
+              <EuiCodeBlock language="json" fontSize="m" paddingSize="m" overflowHeight={300}>
+                {`Modal goes here`}
+              </EuiCodeBlock>
+            </EuiModalBody>
+          </EuiModal>
+        </EuiOverlayMask>
+      )}
+
+      <Panel loading={initLoading}>
+        {initLoading && <Loader data-test-subj="loadingPanelAllRulesTable" overlay size="xl" />}
+
+        <ExceptionsViewerHeader
+          isInitLoading={initLoading}
+          supportedListTypes={availableListTypes}
+          detectionsListItems={detectionsList != null ? detectionsList.totalItems : 0}
+          endpointListItems={endpointList != null ? endpointList.totalItems : 0}
+          onFilterChange={onFiltersChange}
+          onAddExceptionClick={onAddException}
+        />
+
+        {(filterOptions.showEndpointList || filterOptions.showDetectionsList) && (
+          <>
+            <EuiSpacer size="xs" />
+            <StyledText size="s">{exceptionsSubtext}</StyledText>
+          </>
+        )}
+
+        <EuiSpacer size="l" />
+
+        <UtilityBar>
+          <UtilityBarSection>
+            <UtilityBarGroup>
+              <UtilityBarText dataTestSubj="showingRules">
+                {i18n.SHOWING_EXCEPTIONS(pagination.totalItemCount ?? 0)}
+              </UtilityBarText>
+            </UtilityBarGroup>
+
+            <UtilityBarGroup>
+              <UtilityBarAction iconSide="left" iconType="refresh" onClick={onFetchList}>
```

Comment:
Will want to trigger the component loader for the refresh action.

Reply:
Updated in follow up PR!